### PR TITLE
[onert] Introduce RoPE in core ir

### PR DESCRIPTION
**Pull request by ys44kim**
This commit introduces RoPE in core ir

ONE-DCO-1.0-Signed-off-by: youngsik kim <ys44.kim@samsung.com>

issue : https://github.com/Samsung/ONE/issues/14108
draft : https://github.com/Samsung/ONE/pull/14090

### CODE REVIEW

File: runtime/onert/core/src/ir/OperationValidator.cc [hunk 1]
```diff
@@ -451,6 +451,19 @@ void OperationValidator::visit(const operation::Reverse &node)
   OP_REQUIRES(isSameType(output_index, input_index));
 }
 
+void OperationValidator::visit(const operation::RoPE &node)
+{
+  const auto input_index{node.getInputs().at(operation::RoPE::Input::INPUT)};
+  const auto sin_index{node.getInputs().at(operation::RoPE::Input::SIN_TABLE)};
+  const auto cos_index{node.getInputs().at(operation::RoPE::Input::COS_TABLE)};
```
Comment (glistening) [inline runtime/onert/core/src/ir/OperationValidator.cc:458]: As I mentioned before, I don't think `sin` and `cos` tables are mandatory inputs. It is just one possible implementation.

But if we decide to implement and support sin/cos tables only for a while. I am okay.

cc @hseok-oh, @chunseoklee Are you okay in this approach?

Comment (ys44kim) [reply runtime/onert/core/src/ir/OperationValidator.cc:458]: @glistening 
i agree your opinion.
so i hope i keep the current implementation and i will consider generating sin/cos tables by looking at the implementation of toolchain and onert upper layers in the future.
thanks

File: runtime/onert/core/include/ir/operation/RoPE.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_IR_OPERATION_ROPE_H__
+#define __ONERT_IR_OPERATION_ROPE_H__
+
+#include "ir/Operation.h"
+#include "ir/InternalType.h"
```
Comment (zetwhite) [inline runtime/onert/core/include/ir/operation/RoPE.h:21]: May I ask why this header is included?

Comment (ys44kim) [reply runtime/onert/core/include/ir/operation/RoPE.h:21]: i added it wrong by mistake
i will delete it.
thanks

**Review by zetwhite — APPROVED** [review]
LGTM

**Review by ragmani — APPROVED** [review]
LGTM

**Comment by ys44kim** [comment]
@nnfw-bot test-all

**Comment by ys44kim** [comment]
@hseok-oh ping
please check pending

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by hseok-oh** [comment]
@nnfw-bot test onert-cross-aarch64-release onert-cross-debug onert-cross-release

**Comment by ys44kim** [comment]
@glistening 
CI succeeded, Please review

**Review by glistening — APPROVED** [review]
LGTM

**Pull request merged**